components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/autofill/GeckoLoginDelegateWrapper.kt

```diff
@@ -14,18 +16,26 @@ import org.mozilla.geckoview.LoginStorage
  * them to [storageDelegate]. This allows us to avoid duplicating [LoginStorageDelegate] code
  * between different versions of GeckoView, by duplicating this wrapper instead.
  */
-class GeckoLoginDelegateWrapper(private val storageDelegate: LoginStorageDelegate) : LoginStorage.Delegate {
+class GeckoLoginDelegateWrapper(private val storageDelegate: LoginStorageDelegate) :
+    LoginStorage.Delegate {
     override fun onLoginSave(login: LoginStorage.LoginEntry) {
         storageDelegate.onLoginSave(login.toLogin())
     }
 
     override fun onLoginFetch(domain: String): GeckoResult<Array<LoginStorage.LoginEntry>>? {
-        val storedLogins = storageDelegate.onLoginFetch(domain)
+        val result = GeckoResult<Array<LoginStorage.LoginEntry>>()
 
-        return storedLogins
-            .map { it.toLoginEntry() }
-            .toTypedArray()
-            .let { GeckoResult.fromValue(it) }
+        GlobalScope.launch {
+            val storedLogins = storageDelegate.onLoginFetch(domain)
```

Comment:
Nit: we should cancel this job if it is restarted (e.g., someone refreshes in the middle).

```kotlin
var job: Job? = null

override fun onLoginFetch(domain: String): GeckoResult<Array<LoginStorage.LoginEntry>>? {
    ...
    job?.cancel()
    job = GlobalScope.launch {
    ...
}
```

Reply:
This makes sense, yeah!